### PR TITLE
Update/fix typedefs for listr

### PR DESCRIPTION
**Pull request by jgzuke**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/SamVerschueren/listr
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/listr/index.d.ts [hunk 1]
```diff
@@ -11,41 +12,77 @@ import * as stream from "stream";
...
     interface ListrOptions {
         concurrent?: boolean | number;
         exitOnError?: boolean;
-        renderer?: "silent" | "default" | "verbose" | ListrRenderer;
-        nonTTYRenderer?: "silent" | "default" | "verbose" | ListrRenderer;
+        renderer?: ListrRendererValue;
+        nonTTYRenderer?: ListrRendererValue;
+    }
+
+    type ListrContext = any;
```
Comment (polRk) [inline types/listr/index.d.ts]: Do not use any

Comment (polRk) [reply types/listr/index.d.ts]: Maybe generic will be good solution

File: types/listr/index.d.ts [hunk 2]
```diff
@@ -11,41 +12,77 @@ import * as stream from "stream";
...
         title: string;
-        output: any;
+        output: string;
         report(error: Error): void;
         skip(message: string): void;
-        run(ctx?: any): Promise<any>;
+        run(ctx?: ListrContext): Promise<void>;
+    }
+
+    interface ListrError extends Error {
```
Comment (polRk) [inline types/listr/index.d.ts]: Add generic

File: types/listr/index.d.ts [hunk 3]
```diff
@@ -11,41 +12,77 @@ import * as stream from "stream";
...
+    type ListrTaskResult = string | Promise<any> | Listr | stream.Readable | Observable<any>;
+
     interface ListrTask {
+        title: string;
+        task: (ctx: ListrContext, task: ListrTaskWrapper) => void | ListrTaskResult;
+        skip?: (ctx: ListrContext) => void | boolean | string | Promise<boolean>;
+        enabled?: (ctx: ListrContext) => boolean | Promise<boolean> | Observable<boolean>;
+    }
+
+    interface ListrTaskObject extends Observable<ListrEvent> {
```
Comment (polRk) [inline types/listr/index.d.ts]: Add generic, for example:
ListrTaskObject<T = ListrContext > Observable<ListrEvent> {
task: (ctx: T, task: ListrTaskWrapper) => void | ListrTaskResult;

**Review by polRk — CHANGES_REQUESTED** [review]

**Comment by jgzuke** [comment]
@polRk added generic Context type

**Review by polRk — APPROVED** [review]
So well

**Pull request merged**